gobblin-data-management/src/main/java/gobblin/data/management/retention/dataset/CleanableHiveDataset.java

```diff
@@ -0,0 +1,162 @@
+    super(fs, clientPool, table, jobProps, datasetConfig);
+
+    try {
+      this.hiveSelectionPolicy =
+          (VersionSelectionPolicy) GobblinConstructorUtils.invokeFirstConstructor(
+              Class.forName(ConfigUtils.getString(datasetConfig, SELECTION_POLICY_CLASS_KEY, DEFAULT_SELECTION_POLICY_CLASS)),
+              ImmutableList.<Object> of(datasetConfig), ImmutableList.<Object> of(datasetConfig, jobProps), ImmutableList.<Object> of(jobProps));
+
+      this.hiveDatasetVersionFinder =
+          (AbstractHiveDatasetVersionFinder) GobblinConstructorUtils.invokeFirstConstructor(
+              Class.forName(ConfigUtils.getString(datasetConfig, VERSION_FINDER_CLASS_KEY, DEFAULT_VERSION_FINDER_CLASS)),
+              ImmutableList.<Object> of(this.fs, datasetConfig), ImmutableList.<Object> of(this.fs, jobProps));
+    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException | InstantiationException | ClassNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+
+    this.fsCleanableHelper = new FsCleanableHelper(fs, jobProps, datasetConfig, log);
+
+    this.shouldDeleteData = Boolean.valueOf(jobProps.getProperty(SHOULD_DELETE_DATA_KEY, SHOULD_DELETE_DATA_DEFAULT));
+    this.simulate = Boolean.valueOf(jobProps.getProperty(SIMULATE_KEY, SIMULATE_DEFAULT));
+  }
+
+  /**
+   * Drops the partitions selected by {@link #hiveSelectionPolicy}. Also deletes the data associated with it.
+   * <p>
+   * If an {@link Exception} occurs while processing a {@link Partition}, other {@link Partition}s will still be deleted.
+   * However, a {@link RuntimeException} is thrown at the end if there was at least one {@link Exception}.
+   * </p>
+   */
+  @Override
+  public void clean() throws IOException {
+
+    List versions = Lists.newArrayList(this.hiveDatasetVersionFinder.findDatasetVersions(this));
+
+    if (versions.isEmpty()) {
+      log.warn("No dataset version can be found. Ignoring.");
+      return;
+    }
+
+    Collections.sort(versions, Collections.reverseOrder());
+
+    Collection<HiveDatasetVersion> deletableVersions = this.hiveSelectionPolicy.listSelectedVersions(versions);
+
+    Exception ex = null;
+    Set<Path> possiblyEmptyDirectories = new HashSet<>();
+
+    try (AutoReturnableObject<IMetaStoreClient> client = this.clientPool.getClient()) {
+
+      for (HiveDatasetVersion hiveDatasetVersion : deletableVersions) {
+        Partition partition = hiveDatasetVersion.getPartition();
+        try {
+          if (!this.simulate) {
+            client.get().dropPartition(partition.getTable().getDbName(), partition.getTable().getTableName(), partition.getValues(), false);
+            log.info("Successfully dropped partition " + partition.getCompleteName());
+          } else {
+            log.info("Simulating drop partition " + partition.getCompleteName());
+          }
+          if (this.shouldDeleteData) {
```

Comment:
I noticed the simulate mode is in the helper, however, are they using different keys? If so, please make sure hive.simulate forces helper.simulate